lightning/src/ln/channelmanager.rs

```diff
@@ -1973,9 +1973,13 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 					return Ok(())
 				},
 				Err(e) => {
-					// TODO: There is probably a channel manager somewhere that needs to
-					// learn the preimage as the channel may be about to hit the chain.
 					// TODO: Do something with e?
+					// This should only occur if we are claiming an HTLC at the same time as the
+					// HTLC is being failed (eg because a block is being connected and this caused
+					// an HTLC to time out). This should, of course, only occur if the user is the
+					// one doing the claiming (as it being a part of a peer claim would imply we're
+					// about to lose funds) and only if the lock in claim_funds was dropped as a
+					// previous HTLC was failed (thus not for an MPP payment).
```

Comment:
I don't get this new comment. I see the failure where we may try to claim a HTLC off-chain and at same time this or another HTLC is claimed on chain and would trigger a ChannelMonitorUpdateErr, but that's not what this Err path is handling ?

"as it being a part of a peer claim would imply we're about to lose funds" you mean remote timing out onchain a offered HTLC ? but in that case that's not a loss because we won't reveal preimage ?

Reply:
No, I mean locally timing out an HTLC ala #587. Lets continue the conversation there because it requires #587 to make sense.